apps/backoffice-v2/src/lib/blocks/hooks/useDirectorsBlocks/useDirectorsBlocks.tsx

```diff
@@ -129,13 +138,100 @@ export const useDirectorsBlocks = ({
+            const isRevision = decisionStatus === 'revision';
+
+            if (isRevision) {
+              if (workflow?.tags?.includes(StateTag.REVISION)) {
+                const pendingReUploadBlock = createBlocksTyped()
+                  .addBlock()
+                  .addCell({
+                    type: 'badge',
+                    value: <React.Fragment>Pending re-upload</React.Fragment>,
+                    props: {
+                      variant: 'warning',
+                      className: 'min-h-8 text-sm font-bold',
+                    },
+                  })
+                  .build()
+                  .flat(1);
+
+                return pendingReUploadBlock;
+              } else {
+                const reUploadNeededBlock = createBlocksTyped()
+                  .addBlock()
+                  .addCell({
+                    type: 'badge',
+                    value: (
+                      <React.Fragment>
+                        Re-upload needed
+                        <X
+                          className="h-4 w-4 cursor-pointer"
+                          onClick={() =>
+                            removeDecisionById({
+                              documentId: document.id,
+                              contextUpdateMethod: 'director',
+                            })
+                          }
+                        />
+                      </React.Fragment>
+                    ),
+                    props: {
+                      variant: 'warning',
+                      className: `gap-x-1 min-h-8 text-white bg-warning text-sm font-bold`,
+                    },
+                  })
+                  .build()
+                  .flat(1);
+
+                return reUploadNeededBlock;
+              }
+            }
+
+            return undefined;
+          };
+
+          const getReUploadedNeededAction = (
+            decisionStatus: DecisionStatus,
+            workflow: TWorkflowById,
+          ) => {
+            if (decisionStatus !== 'approved' && decisionStatus !== 'revision') {
+              const reUploadNeededBlock = createBlocksTyped()
+                .addBlock()
+                .addCell({
+                  type: 'callToActionLegacy',
+                  value: {
+                    text: 'Re-upload needed',
+                    props: {
+                      revisionReasons: getRevisionReasonsForDocument(document, workflow),
+                      disabled:
+                        (!isDoneWithRevision && Boolean(document.decision?.status)) || noAction,
+                      decision: 'reject',
+                      id: document.id,
+                      contextUpdateMethod: 'director',
+                      workflow,
+                      onReuploadNeeded,
+                    },
+                  },
+                })
+                .build()
+                .flat(1);
+
+              return reUploadNeededBlock;
+            }
+
+            return undefined;
```

Comment:
I understand this is to be explicit but I don't think in the macro `return undefined` adds enough value.

Reply:
This return just shows that undefined expected as well. Im aware that JS functions return undefined by default but this could be not obvious for someone else.

Reply:
Would like this to be a consistent pattern in our monorepo and one we decided on. Same for implicit returns.